### PR TITLE
Updated Stack

### PR DESCRIPTION
**Pull request by shellhub**
### **Describe your change:**

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by shellhub** [comment]
@cclauss Please review. Thanks :)

File: data_structures/stacks/stack.py [hunk 1]
```diff
@@ -29,21 +26,24 @@ def push(self, data):
 
     def pop(self):
         """ Pop an element off of the top of the stack."""
-        if self.stack:
-            return self.stack.pop()
-        else:
+        if self.is_empty():
             raise IndexError("pop from an empty stack")
```
Comment (cclauss) [inline data_structures/stacks/stack.py:30]: Does it do exactly the same thing if you remove lines 29 and 30?

File: data_structures/stacks/stack.py [hunk 2]
```diff
@@ -29,21 +26,24 @@ def push(self, data):
...
+        return self.stack.pop()
 
     def peek(self):
         """ Peek at the top-most element of the stack."""
-        if self.stack:
-            return self.stack[-1]
+        if self.is_empty():
+            raise IndexError("peek from an empty stack")
```
Comment (cclauss) [inline data_structures/stacks/stack.py:36]: Does it do a similar thing if you remove lines 35 and 36?

Comment (shellhub) [reply data_structures/stacks/stack.py:36]: Line3 35 and 36 execute when stack is empty.  `assert()` statements will still be successful if i remove them .

**Review by swapnanildutta — APPROVED** [review]

**Comment by skyloik** [comment]
i will unsub son of a bitch

On 12 Sep 2020 17:54, "Swapnanil Dutta" <notifications@github.com> wrote:

*@swapnanildutta* approved this pull request.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/TheAlgorithms/Python/pull/2414#pullrequestreview-487241466>,
or unsubscribe

**Comment by shellhub** [comment]
Need I update @cclauss ？

**Comment by shellhub** [comment]
Thanks @cclauss . Nice updates. LGTM 👍

**Review by cclauss — APPROVED** [review]

**Pull request merged**